### PR TITLE
Fix start-on-login again

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13377/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13377/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13377/guiReportUpload/serverlog.log

Screenshots:
- [Filenames_that_are_rejected_by_the_server_are_reported.png](https://cache.owncloud.com/public/owncloud/client/13377/guiReportUpload/screenshots/Filenames_that_are_rejected_by_the_server_are_reported.png)

</details>

**Review by TheOneRing — APPROVED** [review]

**Review by TheOneRing — CHANGES_REQUESTED** [review]
Please run clang format and change the title.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
Please fix the commit message

File: src/common/utility_mac.mm [hunk 1]
```diff
@@ -185,53 +185,65 @@
...
-            }
-            NSDictionary *plist = *maybePlist;
-
-            id programValue = plist[@"Program"];
-            if (programValue == nil) {
-                // broken plist, overwrite it
-                auto result = writeNewPlistFile(plistFile, fullPath, enable);
-                if (!result) {
-                    qCWarning(lcUtility) << result.error();
+            if (enable) {
```
Comment (fmoc) [inline src/common/utility_mac.mm:188]: This flag is called `launch` in the function's docstring. This should be corrected/updated (both signature and docstring).

File: src/gui/platform.cpp [hunk 2]
```diff
@@ -20,4 +20,8 @@ Platform::~Platform()
 {
 }
 
+void Platform::migrate()
+{
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/platform.cpp:24]: Why should we log that we did nothing?

File: src/gui/application.cpp [hunk 3]
```diff
@@ -250,6 +250,8 @@ Application::Application(int &argc, char **argv)
     // migrate old configuration files if necessary
     migrateConfigFile(this);
 
+    platform->migrate();
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/application.cpp:253]: In the mac constructor

**Pull request merged**